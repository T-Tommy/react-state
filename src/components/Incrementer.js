import React, { PureComponent } from 'react';

export default class Incremeter extends PureComponent {
  state = {
    number: 1
  };

  handleClick = () => {
    this.setState(state => ({ number: state.number + 1 }));
  };

  render() {
    return (
      <>
        <div>Increment count: { this.state.number }</div>
        <button onClick={ this.handleClick }>
          Increment
        </button>
      </>
    );
  }
}
