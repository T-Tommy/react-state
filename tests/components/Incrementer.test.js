import React from 'react';
import { shallow } from 'enzyme';
import Incrementer from '../../src/components/Incrementer';

describe('Incrementer component', () => {
  it('renders an Incrementer', () => {
    const wrapper = shallow(<Incrementer />);
    expect(wrapper).toMatchSnapshot();
  });

  it('increments on click', () => {
    const wrapper = shallow(<Incrementer />);
    wrapper.find('button').at(0).simulate('click');
    expect(wrapper.state('number')).toEqual(1);
  });
});
