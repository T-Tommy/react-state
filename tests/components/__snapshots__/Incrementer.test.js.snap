// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Incrementer component renders an Incrementer 1`] = `
<Fragment>
  <div>
    Increment count: 
    0
  </div>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
</Fragment>
`;
